### DISCUSSION TITLE
Confused on - Props in React

### DISCUSSION THREAD

**Discussion by AutumnFoto** [nss-day-cohort-47/C47-Discussions#80] [Help Me] [answered]
Kindof a discussion question, kind of a help me...

but with Props, how exactly do we know we are passing in a prop? I know that this syntax is passing a prop - its passing the location id and the location object..
"<LocationCard 
    key={location.id} 
    location={location}"

but is that the only syntax? Because I know lots of things can be passed into props such as functions, variables, objects, ect. So for example functions...are any functions with "export const"..is it implied that its being passed in as a prop to be able to use in another component? 

I hope this makes sense, but basically HOW do we know when something is now a prop...since the word prop appears no where lol??

**Comment by Chadwell-Clark** [comment, marked as answer]
React certainly has thrown a lot at us in a short amount of time. I will attempt a go at this as it helps me to think about it more thoroughly by trying to explain it. If I am wrong then someone will correct me and I will learn from that. 

Props(properties) are sent from a JSX `<element>` (which is inside of a React component) as input to another react component much like an argument is passed to a function.  
In our EmployeeList we map through our employees and then pass each employee object via the `<EmployeeCard>` element... 
```
<EmployeeCard
            employee={employee}
            key={employee.id}
            deleteEmployee={deleteEmployee}
          />
```
... to the EmployeeCard component (function)which receives the properties employee, deleteEmployee and locationList and returns html (JSX) representation of an employee card based partly off the PROPerties passed to it

`export const EmployeeCard = ({ employee, deleteEmployee, locationList }) => {`

the component can use these passed in "props" to read data from and in the case of deleteEmployee cause an action to occur in the parent element when a click event is triggered.
Props are sent from the parent component to the child component and are read only. The child cannot alter the props and the child cannot pass props back to the parent. 

Not sure if that helps any Autumn, but I hope so. Maybe it will help if you hear it in Bryan's voice.

**Reply by BryanNilsen** [reply]
this is an excellent explanation/walkthrough of the code, @Chadwell-Clark
